### ISSUE TITLE
Trigger on user geolocation

### ISSUE THREAD

**Issue by j-rewerts** [CityofEdmonton/IFTTT-Edmonton#95]
This issue will track the work needed to get geolocation triggers working.

Dependant tasks:

- [ ] #61 
- [ ] #60 
- [ ] #59 
- [ ] #74

**Comment by j-rewerts** [comment]
After looking at IFTTT, it seems like most of these aren't possible, as their Location Service is very limited. It only triggers when the user enters or exits a predefined area. This means it isn't suitable for #59 or #74.

For the other 2, which actually do depend on current user location, it likely won't work either. This is because IFTTT has the user set the geofence themselves. For the fruit trees dataset, there are 10000 trees. A user would need to manually set a geofence for every single tree to achieve what we want. 

@WJXHenry Please chime in here.

**Comment by j-rewerts** [comment]
The good news is I'm thinking there's the possibility of doing both #59 and #74 without using IFTTT's Location Service.

**Comment by j-rewerts** [comment]
For the other 2, I found an open source version of IFTTT that looks super cool. It's called [Huginn](https://github.com/huginn/huginn). 🤗 I'm not totally sure this would solve our problem, but it would certainly be worth exploring. The tricky thing is then we would have to publish our own app to get the user's geolocation. Maybe we could look at integrating Huginn with our Open Data portal? Then we could make the portal a progressive web app that gets the user's geolocation.

**Comment by WJXHenry** [comment]
I completely agree with you on #59 working without IFTTT's Location Services, but #74 might be a bit more tricky to tackle since there doesn't seem to be a dataset on parking bans.

I like the idea of using Huginn and having a PWA that we would have so much more control with, which could potentially solve #60 and #61. My main concern would be the amount of time/effort needed to set it all up since we have deadlines? Definitely something we should look into, though.

**Comment by j-rewerts** [comment]
Yeah true, but on the IFTTT side, it seems like it should work if we had access to that dataset in OD.

Agreed on the second part. This is something that would need to be rolled into a larger initiative. I know city staff are investigating improvements to Open Data, so maybe it's a good time to broach something like this?

**Comment by WJXHenry** [comment]
It might be a good thing to introduce the idea to the people in Open Data early. Then if they're on board with it, we could explore further into Huginn.
